pkg/workloadmeta/cached_entity.go

```diff
@@ -0,0 +1,74 @@
+package workloadmeta
+
+import (
+	"sort"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type cachedEntity struct {
+	cached        Entity
+	sources       map[Source]Entity
+	sortedSources []string
+}
+
+func newCachedEntity() *cachedEntity {
+	return &cachedEntity{
+		sources: make(map[Source]Entity),
+	}
+}
+
+func (s *cachedEntity) unset(source Source) bool {
+	if _, found := s.sources[source]; found {
+		delete(s.sources, source)
+		s.computeCache()
+		return true
+	}
+
+	return false
+}
+
+func (s *cachedEntity) set(source Source, entity Entity) bool {
+	_, found := s.sources[source]
+
+	s.sources[source] = entity
+	s.computeCache()
+
+	return found
+}
+
+func (s *cachedEntity) get(source Source) Entity {
+	if source == "" {
+		return s.cached
+	}
+
+	return s.sources[source]
+}
+
+func (s *cachedEntity) computeCache() {
+	var sources []string
+	for source := range s.sources {
+		sources = append(sources, string(source))
+	}
+
+	// sort sources for deterministic merging
```

Comment:
I think we should document (not sure if here or somewhere more visible) that the "priority" of the different sources is established in alphabetical order